src/Picker/ArticlePickerProvider.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+/*
+ * This file is part of Contao.
+ *
+ * Copyright (c) 2005-2017 Leo Feyer
+ *
+ * @license LGPL-3.0+
+ */
+
+namespace Contao\CoreBundle\Picker;
+
+/**
+ * Provides the article picker.
+ *
+ * @author Andreas Schempp <https://github.com/aschempp>
+ */
+class ArticlePickerProvider extends AbstractPickerProvider implements DcaPickerProviderInterface
+{
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getName()
+    {
+        return 'articlePicker';
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getLinkClass()
+    {
+        return 'articles';
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function supportsContext($context)
+    {
+        return 'link' === $context && $this->getUser()->hasAccess('article', 'modules');
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function supportsValue(PickerConfig $config)
+    {
+        return false !== strpos($config->getValue(), '{{article_url::');
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getRouteParameters(PickerConfig $config)
+    {
+        return [
+            'do' => 'article',
+        ];
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getDcaTable()
+    {
+        return 'tl_article';
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getDcaAttributes(PickerConfig $config)
+    {
+        $attributes = ['fieldType' => 'radio'];
+
+        if ('link' === $config->getContext() && $this->supportsValue($config)) {
+            $attributes['value'] = str_replace(['{{article_url::', '}}'], '', $config->getValue());
+        }
+
+        return $attributes;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function convertDcaValue(PickerConfig $config, $value)
```

Comment:
Ok, first, all the `Dca` parts are unnecessary in my opinion, because the class already has `Dca` in it.

But second, the first two methods actually read data from the DCA, so using `Dca` in the method name is kind of correct. The third method, however, reads user input, so …